api/controllers/todoListController.js

```diff
@@ -0,0 +1,45 @@
+const mongoose = require('mongoose');
+const Todo = mongoose.model('Todos')
+
+exports.list_all_todos = (req, res) => {
+  Todo.find({}, (err, todos) => {
+    if(err) {
+      res.send(err);
+    }
+    res.json(todos);
+  });
+};
+
+exports.create_todo = (req, res) => {
+  const new_todo = new Todo(req.body);
```

Comment:
I think it'd be good for them to do validation upfront themselves, like so:

```js
const name = req.body.name;
if (!name) {
  req.status(STATUS_USER_ERROR);
  req.json({ error: 'Must provide a name' });
  return;
}

// after other validation
const newTodo = new Todo({ name, ... });
newTodo.save((err) => {
  if (err) {
    throw err;
  }
  res.json(newTodo);
});
```

This way it's really clear that all validation errors are sent back as 422s, whereas any error from the callback to newTodo.save(...) should be thrown.